### PR TITLE
issue: 1390052 Add Azure machine to Jenkins

### PR DESCRIPTION
**Pull request by serglelv**

### CODE REVIEW

**Comment by serglelv** [comment]
bot:retest

**Comment by serglelv** [comment]
bot:retest

**Comment by serglelv** [comment]
bot:retest

**Comment by serglelv** [comment]
bot:retest

**Comment by serglelv** [comment]
bot:retest

**Comment by serglelv** [comment]
bot:retest

**Comment by serglelv** [comment]
bot:retest

**Comment by serglelv** [comment]
bot:retest

**Comment by liranoz12** [comment]
bot:retest

**Comment by liranoz12** [comment]
@igor-ivanov Please review

File: tests/gtest/sock/sock_socket.cc [hunk 1]
```diff
@@ -101,9 +101,11 @@ TEST_F(sock_socket, ti_4) {
 
 	errno = EOK;
 	fd = socket(PF_INET, SOCK_RAW, IPPROTO_TCP);
+#if 0
 	EXPECT_EQ((sys_rootuser() ? 0 : -1), fd);
-	EXPECT_EQ((sys_rootuser() ? EOK : EPERM), errno);
-
+#endif
```
Comment (igor-ivanov) [inline tests/gtest/sock/sock_socket.cc]: please avoid `#if/endif`. It seems that correct check should be as `EXPECT_EQ((sys_rootuser() ? fd : -1), fd);`

File: tests/gtest/mix/mix_list.cc [hunk 2]
```diff
@@ -124,6 +129,8 @@ TEST_F(mix_list, ti_5) {
 		ASSERT_EQ(reference[ARRAY_SIZE(reference) - i], cur_element->value);
 		i++;
 	}
+#endif
+#endif
```
Comment (igor-ivanov) [inline tests/gtest/mix/mix_list.cc]: what is a reason?

File: tests/gtest/sock/sock_socket.cc (hunk 1 above)
Comment (serglelv) [reply tests/gtest/sock/sock_socket.cc]: Did you test it using sudo?

**Comment by serglelv** [comment]
bot:retest

**Comment by serglelv** [comment]
bot:retest

**Comment by serglelv** [comment]
bot:retest

**Comment by serglelv** [comment]
bot:retest

**Comment by serglelv** [comment]
bot:retest

**Comment by serglelv** [comment]
bot:retest

**Comment by serglelv** [comment]
bot:retest

**Comment by serglelv** [comment]
bot:retest

**Comment by serglelv** [comment]
bot:retest

**Comment by serglelv** [comment]
bot:retest

**Comment by serglelv** [comment]
bot:retest

**Comment by serglelv** [comment]
bot:retest

**Comment by serglelv** [comment]
bot:retest

**Comment by serglelv** [comment]
bot:retest

**Comment by serglelv** [comment]
bot:retest

**Comment by serglelv** [comment]
bot:retest

File: contrib/jenkins_tests/test.sh [hunk 3]
```diff
@@ -29,16 +29,55 @@ if [ $(command -v $test_app >/dev/null 2>&1 || echo $?) ]; then
...
+		sudo scp -q ${test_app} ${rmt_user}@${remote_ip}:${sperf_exec_dir}
+		sudo scp -q ${test_lib} ${rmt_user}@${remote_ip}:${sperf_exec_dir}
+	else
+		sudo -E rsync -q -I -a -r --exclude jenkins --exclude '*.o' --exclude '.deps' --exclude '*.l*' \
+		-e ssh ${WORKSPACE} ${rmt_user}@${remote_ip}:${sperf_exec_dir}
+		sudo scp -q ${test_dir}/sockperf_v2.zip ${rmt_user}@${remote_ip}:${sperf_exec_dir}
+		if [ $? -eq 0 ] ; then
+			subdir=${WORKSPACE##*/}
+			cmd="cd ${sperf_exec_dir}/${subdir} && "
+			cmd+="./autogen.sh && ./configure && make -j4 && "
```
Comment (igor-ivanov) [inline contrib/jenkins_tests/test.sh]: use `$make_opt` from `globals.sh` instead of `make -j4`

File: contrib/jenkins_tests/test.sh [hunk 4]
```diff
@@ -47,15 +86,22 @@ for test_link in $test_ip_list; do
 		test_name=${test_in}-${test}
 		test_tap=${WORKSPACE}/${prefix}/test-${test_name}.tap
 
-		$timeout_exe $PWD/tests/verifier/verifier.pl -a ${test_app} -x " --load-vma=$test_lib " \
-			-t ${test}:tc[1-9]$ -s ${test_ip} -l ${test_dir}/${test_name}.log \
-			-e " VMA_TX_BUFS=20000 VMA_RX_BUFS=20000 " \
-			--progress=0
-
+		if [ ! -z "${test_remote_ip}" ] ; then
```
Comment (igor-ivanov) [inline contrib/jenkins_tests/test.sh]: could you consider just assigning `test_ip=test_remote_ip` under if

File: tests/gtest/mix/mix_list.cc [hunk 5]
```diff
@@ -97,7 +97,14 @@ TEST_F(mix_list, ti_4) {
...
 }
 
+#ifdef __GNUC__
+#include <features.h>
+#if __GNUC_PREREQ(5,4)
+TEST_F(mix_list, DISABLED_ti_5) {
+#else
 TEST_F(mix_list, ti_5) {
+#endif
+#endif
```
Comment (igor-ivanov) [inline tests/gtest/mix/mix_list.cc]: consider moving
```
#ifdef __GNUC__
#include <features.h>
#endif
```
to `gtest/common/def.h`

File: tests/gtest/mix/mix_list.cc [hunk 6]
```diff
@@ -153,7 +160,14 @@ TEST_F(mix_list, ti_7) {
...
 }
 
+#ifdef __GNUC__
+#include <features.h>
+#if __GNUC_PREREQ(5,4)
+TEST_F(mix_list, DISABLED_ti_8) {
+#else
 TEST_F(mix_list, ti_8) {
+#endif
+#endif
```
Comment (igor-ivanov) [inline tests/gtest/mix/mix_list.cc]: see above

File: tests/gtest/sock/sock_socket.cc [hunk 7]
```diff
@@ -45,7 +45,7 @@ class sock_socket : public sock_base {};
  *    Create UDP socket
  * @details
  */
-TEST_F(sock_socket, ti_1) {
+TEST_F(sock_socket, DISABLED_ti_1) {
```
Comment (igor-ivanov) [inline tests/gtest/sock/sock_socket.cc]: please avoid disabling tests for all targets. You can use `sys_rootuser` if it is specific for **root**

**Comment by serglelv** [comment]
bot:retest

**Comment by serglelv** [comment]
bot:retest

**Comment by serglelv** [comment]
bot:retest

**Comment by serglelv** [comment]
bot:retest

**Comment by serglelv** [comment]
bot:retest

**Comment by serglelv** [comment]
@igor-ivanov @liranoz12 Could you please review vlutil.sh ?

File: contrib/jenkins_tests/test.sh [hunk 8]
```diff
@@ -24,21 +24,69 @@ make install
...
+			-e ssh ${WORKSPACE} ${rmt_user}@${test_remote_ip}:${sperf_exec_dir}
+			sudo scp -q ${test_dir}/sockperf_v2.zip ${rmt_user}@${test_remote_ip}:${sperf_exec_dir}
+			if [ $? -eq 0 ] ; then
+				subdir=${WORKSPACE##*/}
+				cmd="cd ${sperf_exec_dir}/${subdir} && "
+				cmd+="./autogen.sh && ./configure && make -j4 && "
+				cmd+="cp src/vma/.libs/*.so ${sperf_exec_dir} &&"
+				cmd+="cd ${sperf_exec_dir} && "
+				cmd+="unzip sockperf_v2.zip && cd sockperf-sockperf_v2 && "
+				cmd+="./autogen.sh && ./configure && make -j4 && cp sockperf ${sperf_exec_dir}"
```
Comment (igor-ivanov) [inline contrib/jenkins_tests/test.sh]: consider using `make $make_opt`

**Comment by serglelv** [comment]
bot:retest

**Comment by serglelv** [comment]
bot:retest

**Comment by serglelv** [comment]
bot:retest

**Comment by serglelv** [comment]
bot:retest

**Comment by serglelv** [comment]
bot:retest

**Comment by serglelv** [comment]
bot:retest

File: contrib/jenkins_tests/test.sh [hunk 9]
```diff
@@ -24,21 +24,69 @@ make install
...
+			sudo scp -q ${install_dir}/sbin/vmad ${rmt_user}@${test_remote_ip}:${sperf_exec_dir}
+			eval "sudo ssh ${rmt_user}@${test_remote_ip} sudo ${sperf_exec_dir}/vmad &"
+		else
+			sudo -E rsync -q -I -a -r --exclude jenkins --exclude '*.o' --exclude '.deps' --exclude '*.l*' \
+			-e ssh ${WORKSPACE} ${rmt_user}@${test_remote_ip}:${sperf_exec_dir}
+			sudo scp -q ${test_dir}/sockperf_v2.zip ${rmt_user}@${test_remote_ip}:${sperf_exec_dir}
+			if [ $? -eq 0 ] ; then
+				subdir=${WORKSPACE##*/}
+				cmd="cd ${sperf_exec_dir}/${subdir} && "
+				cmd+="./autogen.sh && ./configure && make -j4 && "
```
Comment (igor-ivanov) [inline contrib/jenkins_tests/test.sh]: please /-j4/${make_opt}/

**Comment by serglelv** [comment]
@igor-ivanov @liranoz12 Done

**Comment by serglelv** [comment]
bot:retest

**Pull request merged**